lib/cusparse/array.jl

```diff
@@ -256,8 +258,8 @@ SparseArrays.nonzeroinds(g::AbstractCuSparseVector) = g.iPtr
 
 SparseArrays.rowvals(g::CuSparseMatrixCSC) = g.rowVal
 
-LinearAlgebra.issymmetric(M::Union{CuSparseMatrixCSC,CuSparseMatrixCSR}) = false
-LinearAlgebra.ishermitian(M::Union{CuSparseMatrixCSC,CuSparseMatrixCSR}) = false
+LinearAlgebra.issymmetric(M::Union{CuSparseMatrixCSC,CuSparseMatrixCSR}) = size(M, 1) == size(M, 2) ? norm(M - transpose(M), Inf) < 1e-8 : false
+LinearAlgebra.ishermitian(M::Union{CuSparseMatrixCSC,CuSparseMatrixCSR}) = size(M, 1) == size(M, 2) ? norm(M - adjoint(M), Inf) < 1e-8 : false
```

Comment:
Those two methods above don't match LinearAlgebra definition of _exact_ hermitianity/simmetricity in the sense of checking that all elements of `A.== A'`are true.

Also, an arbitrary cutoff of 1E-8 is probably a bad idea, should be made numeric type dependent and possibly have a relative tolerance...

Reply:
Hi @PhilipVinc , 

I agree that the fixed tolerance is not the best choice. I think it can be replaced by the _exact_ definition 

```julia
norm(M - transpose(M), Inf) == 0 # for simmetricity
norm(M - adjoint(M), Inf) == 0 # for hermitianity
```
Indeed, the following code works without introducing the tolerance, and should return the _exact_ definitions

```julia
A = sprand(ComplexF64, 10000, 10000, 0.1)
A2 = CuSparseMatrixCSR(A)

norm(A2 - adjoint(A2), Inf) #2.2140178668261425

A2 += A2'

norm(A2 - adjoint(A2), Inf) #0.0
```

The code written above returns always 0